frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -589,20 +603,34 @@ common_settings.document_metadata_location = {
         genMetadataFolderMenuItem("doc"),
         genMetadataFolderMenuItem("dir"),
         genMetadataFolderMenuItem("hash"),
-        {
-            text = _("Enable hash-based metadata scanning"),
+        { -- hash-based metadata count / TextViewer
+            text = (function()  -- seems like this is running whenever the root menu is opened, not just sdr storage?
```

Comment:
Just use `text_func = function() ... end` :)
Don't call it here, TouchMenu will call it for you.

I think we also need to find a solution to not call DocSettings.getHashDirSdrInfos() twice, in the text_func and the enabled_func (and additionally in the callback if called), but not obvious the way this menu is build, needs some thinking :/

Reply:
> Just use text_func = function() ... end :)
> Don't call it here, TouchMenu will call it for you.

When I do that it errors with `./luajit: frontend/ui/widget/textwidget.lua:181: Wrong text type (expected string)` - I got it working by running the anonymous function to actually get a string value, not a function to be called. But I'm very unfamiliar with Lua, so not sure if that gets run when `common_settings.document_metadata_location` is created (I think when that file/module is initialized at startup, but may even only be when that specific menu is rendered) or someplace else. 

> I think we also need to find a solution to not call DocSettings.getHashDirSdrInfos() twice, in the text_func and the enabled_func (and additionally in the callback if called), but not obvious the way this menu is build, needs some thinking :/

Concur. One option is what if we just check for at least 1 sdr file within hashdocsettings - if one exists then we say that there are hash-based documents. That would be very simple and stay the same even if you have 10000 files in there. When you actually tap to show them, _then_ it does the full calculation.

Reply:
Maybe keep `No documents with hash-based metadata` the same, but instead of `Show %1 documents with hash-based metadata`, write something like "There are documents with hash-based metadata, tap to find/display them". I still don't know how to make ⚠ dynamic but that solves the performance issue. Only thing is I can't figure out a shorter description for the above to fit in one settings 'row', not sure if they can be multiline or if there's a way to completely show/hide a secondary button for "Find documents with hash-based metadata" (not just grey out).

Reply:
Example impl at https://github.com/koreader/koreader/pull/10945#issuecomment-1743469425

Reply:
> I still don't know how to make ⚠ dynamic but that solves the performance issue

Well somehow it _is_ dynamic now after following what @hius07 said, not sure what changed it.